packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -712,13 +712,12 @@ function createInstances(cluster: DatabaseClusterNew, props: DatabaseClusterBase
       deleteAutomatedBackups: props.instanceProps.deleteAutomatedBackups,
     });
 
-    // If removalPolicy isn't explicitly set,
-    // it's Snapshot for Cluster.
-    // Because of that, in this case,
-    // we can safely use the CFN default of Delete for DbInstances with dbClusterIdentifier set.
-    if (props.removalPolicy) {
-      applyRemovalPolicy(instance, props.removalPolicy);
-    }
+    // The RemovalPolicy on the cluster is SNAPSHOT by default, or explicitly configured.
+    // We can safely default to DESTROY for instances that are part of a cluster.
+    //
+    // TODO: This should probably be `DESTROY` always, no, instead of inheriting the
+    // cluster removal policy? It would lead to useless additional snapshots or retentions...
```

Comment:
Did you see the comment above, that you removed?

The default for `DbInstance`s is `SNAPSHOT` if `DBClusterIdentifier` is not set, but `DELETE` if `DBClusterIdentifier` is set (like it is here). So no, no useless additional snapshots will be generated. Maybe leave that comment in place?

Docs: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html

Reply:
Yes, but since we're making a `DBInstance` here as part of a `DBCluster`, the cluster identifier will always be set, right? So we fall into one branch of that statement, where the CFN default would be `DELETE` (and that's the same default we set).

However, what also happens now (because we reuse `props.removalPolicy`), is that if `SNAPSHOT` or `RETAIN` are set on the cluster, they are **also set on the DBInstance**. This is pointless -- if the cluster is already snapshotted or retained, we don't need to snapshot or retain the individual instances as well, do we?

Reply:
We definitely need to `RETAIN` if Cluster is `RETAIN` (otherwise, the instances themselves will be deleted, even if if the Cluster is retained, making the Cluster useless).

Reply:
If `props.removalPolicy` is `SNAPSHOT`, we can probably leave it as `DELETE` on the instance.

Reply:
However, the unnecessary `SNAPSHOT` on the instance is the behavior today, so if you want to leave it as-is in this PR, I'm fine with it.

The `RETAIN` has to be changed though.

Reply:
Yeah we have to look at this in aggregate.

Let me see if I understand correctly:

| If the Cluster policy is... | ...then the Instance policy has to be | Notes |
|----------|--------------|--------------|
| Delete | Delete | This one is easy |
| Snapshot | Delete | Right? |
| Retain | Retain | Right? |

Reply:
Okay did that.